tool/tsh/tsh.go

```diff
@@ -625,6 +627,17 @@ func Run(args []string, opts ...cliOption) error {
 	reqReview.Flag("deny", "Review proposes denial").BoolVar(&cf.Deny)
 	reqReview.Flag("reason", "Review reason message").StringVar(&cf.ReviewReason)
 
+	// TODO(nic): unhide this command when the rest of search-based access
+	// requests is implemented (#10887)
+	reqSearch := req.Command("search", "Search for resources to request access to").Hidden()
```

Comment:
Don't want to question UX if it's already been approved by the product team but I do wonder if integrating this functionality into `tsh ls` instead of a new command has been considered (e.g. `tsh ls --kinds xxx --search yyy`, I think it already has some filter flags also).

Reply:
I haven't put too much thought into that. Looks like `tsh ls` so far only deals with ssh nodes, to add a `--kinds` flag kind of breaks the semantics I think?

The alternative would be to add a `--request` flag to every `ls` command, `tsh apps ls`, `tsh db ls`, etc. Not really sure if that is better.